code/modules/mob/living/living.dm

```diff
@@ -1566,39 +1566,71 @@
 	var/mob/living/U = user
 	if(isliving(dropping))
 		var/mob/living/M = dropping
-		if(M.can_be_held && U.pulling == M)
+		//If they can be picked up, and we can't (prevents recursion), try to pick the target mob up as an item.
+		if(U.pulling == M && (HAS_TRAIT(M, TRAIT_HOLDABLE) && !HAS_TRAIT(src, TRAIT_HOLDABLE)))
 			M.mob_try_pickup(U)//blame kevinz
 			return//dont open the mobs inventory if you are picking them up
 	. = ..()
 
-/mob/living/proc/mob_pickup(mob/living/L)
-	var/obj/item/clothing/head/mob_holder/holder = new(get_turf(src), src, held_state, head_icon, held_lh, held_rh, worn_slot_flags)
-	L.visible_message(span_warning("[L] scoops up [src]!"))
-	L.put_in_hands(holder)
-
-/mob/living/proc/set_name()
-	numba = rand(1, 1000)
-	name = "[name] ([numba])"
-	real_name = name
-
-/mob/living/proc/mob_try_pickup(mob/living/user, instant=FALSE)
-	if(!ishuman(user))
+/**
+ * Attempt to pick up a mob.
+ *
+ * `src` is the mob being picked up, `user` is the mob doing the pick upping.
+ *
+ * Arguments:
+ * * mob/living/user - The user attempting to pick up this mob.
+ * * instant - Should the mob be picked up instantly?
+ */
+/mob/living/proc/mob_try_pickup(mob/living/user, instant = FALSE)
+	. = FALSE
+	if(!mob_pickup_checks(user))
 		return
+	if(!instant)
+		user.visible_message(span_warning("[user] starts trying to pick up [src]!"),
+			span_danger("You start trying to pick up [src]..."), ignored_mobs = src)
+		to_chat(src, span_userdanger("[user] starts trying to pick you up!"))
+		if(!do_after(user, 2 SECONDS, src))
+			return
+		if(!mob_pickup_checks(user)) // Check everything again after the timer
+			return
+	mob_pickup(user)
+	return TRUE
+
+/mob/living/proc/mob_pickup_checks(mob/living/user, display_messages = TRUE)
```

Comment:
make it like Baystation where you need to be 1 mob_size class smaller to be picked up by something

Reply:
Rather than `TRAIT_HOLDABLE`, or..?

Reply:
Yes.